storops/unity/resource/host.py

```diff
@@ -311,6 +326,40 @@ def delete_ip_port(self, address):
                      .format(address, self.name))
         return resp
 
+    def update_initiators(self, iqns=None, wwns=None):
+        """Primarily for puppet-unity use.
+
+        Update the iSCSI and FC initiators if needed.
+        """
+        # first update the iqns
+        iqns = iqns if iqns else []
+        current_iqns = []
+        if self.iscsi_host_initiators:
+            current_iqns = [initiator.initiator_id
+                            for initiator in self.iscsi_host_initiators]
+        to_add = set(iqns) - set(current_iqns)
+        to_delete = set(current_iqns) - set(iqns)
+
+        for a in to_add:
+            self.add_initiator(a)
+        for d in to_delete:
+            self.delete_initiator(d)
+
+        # Then update the wwns
+        wwns = wwns if wwns else []
+        current_wwns = []
+        if self.fc_host_initiators:
+            current_wwns = [initiator.initiator_id
+                            for initiator in self.fc_host_initiators]
+
+        to_add = set(wwns) - set(current_wwns)
+        to_delete = set(current_wwns) - set(wwns)
+
+        for a in to_add:
+            self.add_initiator(a)
+        for d in to_delete:
```

Comment:
Please consider separate the update logic into a class.  For example, a class may have two input, `prev` and `now`.  The output are `to_add` and `to_remove`.